### ISSUE TITLE
SSL offload does not work after jetty upgrade (rundeck > 2.6) 

### ISSUE THREAD

**Issue by Dalma-TB** [rundeck/rundeck#2509]
Issue type: Support

**My Rundeck detail**

* Rundeck version: 2.8.2
* install type: (rpm,deb,war,launcher?) deb
* OS Name/version: Ubuntu 16.04
* DB Type/version: mysql? postgres? h2?.. mysql

**Expected Behavior**
Correct redirect to login page.

**Actual Behavior**
The setup is: AWS Loadbalancer with ssl offloading, nginx, rundeck. The configuration was working on 2.6, after jetty upgrade it stopped working.
For a metrics/ping request I get the pong via nginx.
If the `-Drundeck.jetty.connector.forwarded=true` is set in profile, then the login page gets a redirect loop with the very same URL, even the protocol is the same. Without, it is a simple redirect from https to http.
The grails.serverUrl is set to https://r...
Basically a similar issue to #2490.

**How to reproduce Behavior**
Version 2.6, without the jetty option works perfect. After upgrading, main page cannot be loaded with the setup defined above.

**Comment by gschueler** [comment]
i'm curious what the requests look like at the rundeck server. Are you able to see the `x-forwarded-*` headers?

**Comment by Dalma-TB** [comment]
Nginx gets all the X-Forwarded-Proto, X-Forwarded-For and X-Forwarded-Port headers. 
In the proxy settings the header is explicitly set:
`proxy_set_header X-Forwarded-Proto $http_x_forwarded_proto;`, but without this, the behaviour is exactly the same.
Not sure how to check what rundeck actually sees from this. 
If I do a curl -v -H "X-Forwarded-Proto: https" http://localhost:4440/<mountpoint>, then the location will have the redirect to https://localhost:4440/<mountpoint>. When nginx sends the same, the location is https://<serverurl>/<mountpoint>/user/login;jsessionid=<same-for-the-loop>.

**Comment by Dalma-TB** [comment]
How can I see those headers in rundeck itself? I have tried several settings in log4j.properties but the only entry is still the url in the access log.

**Comment by gschueler** [comment]
@Dalma-TB you can try turning on debug logging in jetty:
`-Dorg.eclipse.jetty.server.Server.LEVEL=ALL`

or DEBUG level for everything (pretty verbose):

`-Dorg.eclipse.jetty.LEVEL=DEBUG`

or just for the http parsing:

`-Dorg.eclipse.jetty.http.LEVEL=DEBUG`

e.g.:

    java -Xmx1g -Dorg.eclipse.jetty.server.Server.LEVEL=ALL -jar rundeck-launcher-2.8.2.jar

**Comment by Dalma-TB** [comment]
Maybe I'm doing something wrong, because with these options the log is definitely more verbose, but do not report the actual headers. Anyway, after having some security updates (libgnutls, libc6),  I cannot reproduce the issue anymore. (Nothing else has changed besides this.) Thank you anyway.

**Comment by gschueler** [comment]
@Dalma-TB do you mean you were able to get it working again?

**Comment by Dalma-TB** [comment]
Yes, without changing anything besides the OS security updates, now it suddenly works.
